src/net.cpp

```diff
@@ -1592,6 +1595,10 @@ void ThreadOpenConnections()
             if (IsLimited(addr))
                 continue;
 
+            // only connect to full nodes
```

Comment:
> Why only connect to full nodes? Surely this is only a requirement when we
are more than 550 blocks behind on the blockchain.

No, we always want this requirement.

Otherwise we may end up trying to connect to lightweight clients that do
not even relay blocks at all.